demo/full/bxf-parser2/tmp/hapi.js

```diff
@@ -0,0 +1,417 @@
+      const privateLength = be4toi(initData, position);
+      position += 4;
+      const privateData = initData.subarray(position, position + privateLength);
+      acc.push(privateData);
+    }
+    return acc;
+  }, []);
+  const widevinePssh = psshs.find((pssh) => pssh.length === 20);
+  if (widevinePssh) {
+    return bytesToHex(widevinePssh.subarray(4, 20), "-");
+  }
+  return undefined;
+}
+
+parseBXF(BXFUrl, BXFParser, "text").then(async ({ contents: parsedContents }) => {
+  console.log("############## Parsed BXF ################");
+  const contents = [];
+  const overlays = [];
+  debugger;
+  for (const parsedContent of parsedContents) {
+    const videos = parsedContent.video;
+    if (videos) {
+      for(const video of videos) {
+        if (
+          video.type === "CIN" ||
+          video.type === "MUS" ||
+          video.type === "CA" ||
+          video.type === "SA" ||
+          video.type === "EMI" ||
+          video.type === "SER"
+        ) {
+          try {
+            const { manifestURL, licenseURL } = await getManifestURL(token, video.affaire, video.pgrm);
+            contents.push({
+              name: video.title,
+              url: manifestURL + "/Manifest",
+              startTime: video.startTime,
+              endTime: video.endTime,
+              transport: "smooth",
+            });
+            const logos = parsedContent.logo;
+            if (logos) {
+              logos.forEach((logo) => {
+                const { title, offset } = logo;
+                const url = getURLForLogoTitle(title);
+                if (url) {
+                  const { begin, end } = offset;
+                  overlays.push({
+                    start : video.startTime + (begin || 0),
+                    end : video.endTime - (end || 0),
+                    timescale : 1,
+                    version : 1,
+                    elements : [{
+                      url,
+                      format : "png",
+                      xAxis : "0%",
+                      yAxis : "0%",
+                      height : "100%",
+                      width : "100%",
+                    }],
+                  });
+                }
+              });
+            }
+            await getKID(manifestURL).then(({ kid }) => {
+              kidByAffairePgrm[kid] = licenseURL;
+            });
+          } catch(_) {
+          }
+        }
+      }
+    }
+  }
+
+  console.log("############## Built Metaplaylist ################");
+
+  const contentsWithBlack = [];
+
+  contents.forEach((content, i) => {
+    if (i === 0) {
+      contentsWithBlack.push(content);
+    } else {
+      if (content.startTime > contents[i -1].endTime) {
+        contentsWithBlack.push({
+          url: blackManifestURL,
+          endTime: content.startTime,
+          startTime: contents[i -1].endTime,
+          transport: "dash",
+        });
+        contentsWithBlack.push(content);
+      }
+    }
+  });
+
+  debugger; 
```

Comment:
![Vulnerability](https://sonarsource.github.io/sonar-github/vulnerability.png 'Vulnerability') Vulnerability: Remove this debugger statement. ([javascript:DebuggerStatement](https://sonarcloud.io/organizations/rx-player/rules#rule_key=javascript%3ADebuggerStatement))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=366&issues=AWbt7Z3fW3Y69bz5aGW8&open=AWbt7Z3fW3Y69bz5aGW8)

[ISSUE_KEY]: AWbt7Z3fW3Y69bz5aGW8